PayPalDataCollector/src/main/java/com/paypal/android/paypaldatacollector/PayPalDataCollector.kt

```diff
@@ -0,0 +1,73 @@
+package com.paypal.android.paypaldatacollector
+
+import android.content.Context
+import android.util.Log
+import com.paypal.android.core.CoreConfig
+import com.paypal.android.core.Environment
+import lib.android.paypal.com.magnessdk.InvalidInputException
+import lib.android.paypal.com.magnessdk.MagnesSDK
+import lib.android.paypal.com.magnessdk.MagnesSettings
+import lib.android.paypal.com.magnessdk.MagnesSource
+
+/**
+ * Enables you to collect data about a customer's device and correlate it with a session identifier on your server.
+ */
+class PayPalDataCollector internal constructor(
+    coreConfig: CoreConfig,
+    private val magnesSDK: MagnesSDK,
+    private val uuidHelper: UUIDHelper
+) {
+
+    private val environment = when(coreConfig.environment) {
+        Environment.LIVE -> lib.android.paypal.com.magnessdk.Environment.LIVE
+        Environment.STAGING -> lib.android.paypal.com.magnessdk.Environment.STAGE
+        Environment.SANDBOX -> lib.android.paypal.com.magnessdk.Environment.SANDBOX
+    }
+
+    constructor(coreConfig: CoreConfig) : this(
+        coreConfig,
+        MagnesSDK.getInstance(),
+        UUIDHelper()
+    )
+
+    /**
+     * Gets a Client Metadata ID at the time of payment activity. Once a user initiates a PayPal payment
```

Comment:
This will be used for other payments than just `PayPal` in the future right? Maybe we should make this more generic. Also maybe when referring to `PayPal` as the processor we should say the SDK or the server depending? Seems a little confusing

Reply:
im honestly not sure, doesnt the whole SDK fall under PayPal? From the merchant side, im guessing that Northstar is PayPal